src/lib/y2network/connection_config/base.rb

```diff
@@ -204,6 +204,23 @@ def static?
         bootproto ? bootproto.static? : true
       end
 
+      # Return the first hostname associated with the primary IP address
+      #
+      # @return [String, nil]
+      def hostname
+        hostnames&.first
+      end
+
+      # Convenience method in order to modify the canonical hostname mapped to
+      # the primary IP address.
+      #
+      # @param name [String] hostnamme mapped to the primary IP address
+      def hostname=(name)
+        short_name = name&.split(".")&.first
```

Comment:
according to doc name should not be nil. And if it is not nil, then many magic is removed from this method. ( maybe just check it and raise argument error if you want to be super sure. )